design/addon-categories.md

```diff
@@ -0,0 +1,132 @@
+
+## Goals
+
+- Offer high-level guidance to authors of kURL specs about which add-ons to include in their installer.
+- Decrease the likelihood of designing kURL specs that result in non-functioning installations.
+
+## Non Goals
+
+- Detecting syntax and low-level errors within the configurations of individual add-ons.
+
+## Background
+
+Authors of kURL specs must select from a growing list of add-ons.
+They may not be familiar with the function of the add-on and may not be aware of its dependencies.
+Authors must spend time researching each add-on and try installations to verify their spec results in a working cluster.
+
+A better experience would be to provide guidance that would lead to working clusters by default.
+
+## High-Level Design
+
+Each add-on belongs to one or more of the following categories:
+
+- "Container Runtime"
+- "CNI Plugin"
+- "PVC Provisioner"
+- "Object Store"
+- "Ingress"
+- "Logs"
+- "Snapshots"
+- "Metrics & Monitoring"
+- "Cluster Administration"
+- "Application Management"
+
+Each add-on can specify dependencies on other categories.
+
+Add-ons belonging to the same category conflict with each other.
+
+## Detailed Design
+
+Add a GET /add-ons endpoint that returns a list of add-ons along with their dependencies.
+Example response:
+
+```
+[
+	{
+		name: "docker",
+		fulfills: ["Container Runtime"],
+		requires: [],
+	},{
+		name: "containerd",
+		fulfilles: ["Container Runtime"],
+		requires: [],
+	},{
+		name: "weave",
+		fulfills: ["CNI Plugin"],
+		requires: ["Container Runtime"],
+	},{
+		name: "calico",
+		fulfills: ["CNI Plugin"],
+		requires: ["Container Runtime"],
+	},{
+		name: "rook",
+		fulfills: ["PVC Provisioner", "Object Store"],
+		requires: ["CNI Plugin", "Container Runtime"],
+	},{
+		name: "openEBS",
+		fulfills: ["PVC Provisioner"],
+		requires: ["CNI Plugin", "Container Runtime"],
+	},{
+		name: "minio",
+		fulfills: ["Object Store"],
+		requires: ["CNI Plugin", "Container Runtime"],
+	},{
+		name: "contour",
+		fulfills: ["Ingress"],
+		requires: ["CNI Plugin", "Container Runtime"],
+	},{
+		name: "fluentd",
+		fulfills: ["Logs"],
+		requires: ["CNI Plugin", "Container Runtime"],
+	},{
+		name: "kotsadm",
+		fulfills: ["Application Management"],
+		requires: ["CNI Plugin", "Container Runtime", "PVC Provisioner", "Object Store", "Registry", "Snapshots"],
+	},{
+		name: "registry",
+		fulfills: ["Registry"],
+		requires: ["CNI Plugin", "Container Runtime","Object Store"],
+	},{
+		name: "velero",
+		fulfills: ["Snapshots"],
+		requires: ["CNI Plugin", "Container Runtime", "Object Store"],
+	},{
+		name: "prometheus",
+		fulfills: ["Metrics & Monitoring"],
+		requires: ["CNI Plugin", "Container Runtime", "PVC Provisioner"],
+	},{
+		name: "ekco",
+		fulfills: ["Cluster Administration"],
+		requires: ["CNI Plugin", "Container Runtime"],
+	}
+]
+```
+
+## Alternatives Considered
+
+### Recommends
+
+Each add-on's `requires` list could be separated into `recommends` and `requires`.
```

Comment:
I think this is a good direction to go. Take in to consideration this design:

<img width="1311" alt="Screen Shot 2020-05-05 at 4 28 59 PM" src="https://user-images.githubusercontent.com/4110866/81118007-b26b6f00-8eed-11ea-8679-811b56cbe7b2.png">


I think there would be a lot of value in the "Works well with" section. To do this we could use the `recommends` field to grab items out of the suggested categories.

Reply:
Moved to the design.